zerver/lib/url_validator.py

```diff
@@ -0,0 +1,36 @@
+'''
+This module defines some custom url validation regexes/utilities.
+'''
+
+def get_uncompiled_netloc_re() -> str:
+    '''
+    From: https://docs.djangoproject.com/en/3.0/_modules/django/core/validators/#URLValidator
+
+    This method takes code from Django's URLValidator. We cannot use the URLValidator as is
+    because it performs checks on the scheme as well and we have code to deal with schemes
+    already that doesn't agree with URLValidator. Thus, we extract the netloc specific parts
+    of URLValidator here.
```

Comment:
Do we really have to? The upstream `URLValidator` constructor accepts an optional list of schemes, so I’d think we should be able to use it directly.